lib/ddtrace/encoding.rb

```diff
@@ -43,6 +43,34 @@ def content_type
       def encode(obj)
         JSON.dump(obj)
       end
+
+      # New version of JSON Encoder that is API compliant.
+      module V2
+        extend JSONEncoder
+
+        ENCODED_IDS = [
+          :trace_id,
+          :span_id,
+          :parent_id
+        ].freeze
+
+        module_function
+
+        def encode_traces(traces)
+          trace_hashes = traces.collect do |trace|
+            # Convert each trace to hash
+            trace.map(&:to_hash).tap do |spans|
+              # Convert IDs to hexadecimal
+              spans.each do |span|
+                ENCODED_IDS.each { |id| span[id] = span[id].to_s(16) }
```

Comment:
spans will *always* have these ids? we don't need to do a `if span.key?(id)` or anything?

Reply:
They are fields of `Span`, but I think this is a fair point: perhaps at some point they will be removed. This would probably add a key then that shouldn't be there. Will add this check... good catch.